File: example/esbuild.config.go
package main

import (
	devserver "github.com/Falldot/esbuild-dev-server"
	"github.com/evanw/esbuild/pkg/api"
)

func main() {
	devserver.Start(
		api.Build(api.BuildOptions{
			EntryPoints: []string{"src/index.js"},
			Outdir:      "dist",
			Incremental: true,
			// and more options ...
		}),
		devserver.Options{
			Port:            "8080",            // optional, default: 8080
			WatchDir:        "src",             // optional, default: "src"
			Index:           "dist/index.html", // optional
			StaticDir:       "dist",            // optional
			OnBeforeRebuild: func() {},         // optional
			OnAfterRebuild:  func() {},         // optional
		},
	)
}

File: example/esbuild.config.js
const {build} = require("esbuild")
const esBuildDevServer = require("esbuild-dev-server")

esBuildDevServer.start(
	build({
		entryPoints: ["src/index.js"],
		outdir: "dist/js",
		incremental: true,
		// and more options ...
	}),
	{
		port:      "8080", // optional, default: 8080
		watchDir:  "src", // optional, default: "src"
		index:     "dist/index.html", // optional
		staticDir: "dist", // optional
		onBeforeRebuild: {}, // optional
		onAfterRebuild:  {}, // optional
	}
)